starlette_context/middleware/mixin.py

```diff
@@ -0,0 +1,56 @@
+import logging
+from typing import Awaitable, Callable, Optional, Union
+
+from starlette_context.errors import StarletteContextException
+from starlette.responses import PlainTextResponse, Response
+from starlette.requests import Request
+
+
+class StarletteContextMiddlewareMixin:
+    def __init__(
+        self,
+        error_handler: Optional[
+            Callable[
+                [Request, StarletteContextException],
+                Union[Response, Awaitable[Response]],
```

Comment:
`         return await self.error_handler(request, exc)`
that looks very much like only awaitables are supported. why the union?
(and i'd rather support only one than digging into the can of worms of supporting either)